src/main/java/com/revature/services/QuestionEvalLogicImpl.java

```diff
@@ -0,0 +1,165 @@
+package com.revature.services;
+
+import com.revature.domain.QuestionComment;
+import com.revature.domain.QuestionEval;
+import com.revature.repositories.EvalRepository;
+import com.revature.repositories.QuestionCommentRepository;
+import com.revature.repositories.QuestionEvalRepository;
+import com.revature.validation.JsonValidation;
+import com.revature.validation.exceptions.NotFoundException;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.validation.ConstraintViolationException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional(readOnly=false, isolation=Isolation.READ_COMMITTED)
+public class QuestionEvalLogicImpl implements QuestionEvalLogic{
+	
+	@Autowired
+	private QuestionEvalRepository dao;
+	
+	@Autowired
+	private QuestionCommentRepository commentDao;
+	
+	@Autowired
+	private EvalRepository evalDao;
+
+    @Autowired
+    private QuestionLogic questionLogic;
+
+    @Autowired
+    private JsonValidation validation;
+	
+	@PersistenceContext
+    private EntityManager entityManager;
+	
+//CREATE-----------------------------------
+
+    @Override
+	@Transactional
+	public QuestionEval createQuestionEval(QuestionEval qEval, Integer evalId) {
+        validation.validateQuestionEvaluationFieldsExistingEval(qEval);
+		
+		qEval.setEval(evalDao.findOne(evalId));
+        if (qEval.getEval() == null) {
+            throw new ConstraintViolationException("Evaluation with id " + evalId + " does not exist", null);
+        }
+
+		// Set eval property of comments so cascade works
+		if (qEval.getComments() != null) {
+			qEval.getComments().forEach((comment) -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "comment" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1611)